pallene/coder.lua

```diff
@@ -1437,6 +1437,42 @@ gen_cmd["BuiltinIoWrite"] = function(self, cmd, _func)
     return util.render([[ pallene_io_write(L, $v); ]], { v = v })
 end
 
+gen_cmd["BuiltinMathAbs"] = function(self, cmd, _func)
+    local dst = self:c_var(cmd.dsts[1])
+    local v = self:c_value(cmd.srcs[1])
+    return util.render([[ $dst = l_mathop(fabs)($v); ]], { dst = dst, v = v })
+end
+
+gen_cmd["BuiltinMathExp"] = function(self, cmd, _func)
+    local dst = self:c_var(cmd.dsts[1])
+    local v = self:c_value(cmd.srcs[1])
+    return util.render([[ $dst = l_mathop(exp)($v); ]], { dst = dst, v = v })
+end
+
+gen_cmd["BuiltinMathLog"] = function(self, cmd, _func)
+    local dst = self:c_var(cmd.dsts[1])
+    local v = self:c_value(cmd.srcs[1])
+    local b = self:c_value(cmd.srcs[2])
+	local base_num = tonumber(b)
```

Comment:
I believe that converting the C string back to a number technically works, but it feels fragile because the c_value doesn't promise that the string it produces is something that can be passed to `tonumber`. For example, a future version of `c_value` could return a string that contains a `/**/` comment.

I think a better way to do this would be to check the `._tag` field of the `cmd.srcs[2]`. If it is an `ir.Value.Float` then we know it is a floating-point constant and we can look at the `.value` to find out what value it has.